docs/environment.md

```diff
@@ -12,28 +12,85 @@ You have two options for working with our codebase:
 That's because PyTorch has better support for Linux.  If you want to use WSL2, please follow
 [these instructions](/docs/WSL.md) , that correspond to the manual installation in the official docs.
 
-## Installing Conda or Miniconda
+### Install Conda or Miniconda
 You can skip this step if you have installed WSL as per the previous item.
 
 Download a Conda or Miniconda [installer for your platform](https://docs.conda.io/en/latest/miniconda.html)
 and run it.
 
-## Creating a Conda environment
-Note that in order to create the Conda environment you will need to have build tools installed on your machine. If you are running Windows, they should be already installed with Conda distribution.
+## Create a Conda Environment
+
+There are three important configuration files in this repo for creating conda environments:
+- **`primary_deps.yml`** - This file contains the list of primary package dependencies, and can be used to create an environment on any OS.
+- **`environment.yml`** - **DO NOT EDIT THIS FILE MANUALLY**. This file contains a locked list of primary and secondary dependencies that is used to create the environments for your AzureML jobs and local Linux deployments. As such it contains linux-specific platform dependencies (see [Set Up WSL](#set-up-windows-subsystem-for-linux)) and cannot be used to create a windows environment.
+- **`environment_win.yml`** - This file contains a windows-dependent list of primary and secondary dependencies. This is the file used to create a locked python environment on windows machines.
+
+### Create environment from lockfile
+
+To create an environment from one of the lockfiles, run the following command:
+
+```conda env create --file <environment.yml/environment_win.yml>```
+
+Selecting the relevant file depending on your OS platform (`environment.yml` for Linux, `environment_win.yml` for windows). This will create a new conda environment which can be activated by running:
+
+```conda activate InnerEye```
+
+### Upgrade primary dependency / create new `environment.yml`
+
+**Linux/AML Users**
+1. Make your desired changes in `primary_deps.yml`. Make sure your package name and version are correct.
+2. To create a new environment and a valid `environment.yml`, run the following command:
+
+    ```bash -i create_and_lock_environment.sh```
+
+3. Voila! You will now have a new conda environment with your desired primary package versions, as well as a new `environment.yml` which can be ingested by AzureML to create a copy of your local environment.
+
+**Windows Users**
+
+NOTE: any changes to packages made here will not be repeated in any AML deployments. To edit package versions to be used in AML (i.e. to generate a new lockfile `environment.yml`) you will need to follow the steps on creating an conda env in Linux (previous section). If you do not have access to a Linux machine, we highly recommened using [WSL](#set-up-windows-subsystem-for-linux).
```

Comment:
Maybe re-phase to first focus on what's possible? Something like "If you are using the InnerEye toolbox in a native WIndows environment (but not in the Windows subsystem for Linux WSL), you need to follow the steps below. If you are submitting jobs to AzureML from your Windows Python environment, you will need to do what's described above in the Linux/AzureML section)